### DISCUSSION TITLE
Sending data to core-data through REST API from Device Service

### DISCUSSION THREAD

**Discussion by sudhamani-hcl** [edgexfoundry/community#101] [General]
Hi Team,

We are writing custom device service, In which we wanted to post data to core-data service through REST API.

From the reference https://docs.edgexfoundry.org/2.3/microservices/core/data/Ch-CoreData/ 
It is saying that we can send data to message bus or directly to core-data through REST API's.

What is the configuration I need to do to send device data to core-data through REST?
or
Do I need to handle this in the device service code itself?

Any help would be appreciated!!

And also could you please share any reference device services using REST for sending data to core-data service?

Thanks,
Sudhamani

**Comment by cloudxxx8** [comment]
We handle the event sending logic in SDK, and we use the message bus only in SDK after v3.0
https://github.com/edgexfoundry/device-sdk-go/blob/cc1c191f7136a36b2a7f5a8b6d3c5c2ccaccd9b2/internal/common/utils.go#L52

Please see this change
https://github.com/edgexfoundry/device-sdk-go/commit/482e5b9a17b7b3fb4ec31d9b3676b675d900f01c

In v2.x images, you can choose to use REST call by setting `Device.UseMessageBus` configuration to `false` in any device service.
https://github.com/edgexfoundry/device-virtual-go/blob/0c2f1e9213ddf10d1b3316c025c729eab3a2bc23/cmd/res/configuration.toml#L126
https://github.com/edgexfoundry/device-sdk-go/blob/b3f594a181df4350d50f9cefe1b70e34eb008ea7/internal/common/utils.go#L84-L104

After v3.0, if you still have to use REST call to send Events, you need to create your own customized Device Service.
